comps/dataprep/opensearch/README.md

```diff
@@ -0,0 +1,283 @@
+
+- option 1: Install Single-process version (for 1-10 files processing)
+
+```bash
+apt update
+apt install default-jre
+apt-get install tesseract-ocr -y
+apt-get install libtesseract-dev -y
+apt-get install poppler-utils -y
+# for langchain
+cd langchain
+pip install -r requirements.txt
+```
+
+<!-- - option 2: Install multi-process version (for >10 files processing)
+
+```bash
+cd langchain_ray; pip install -r requirements_ray.txt
+``` -->
+
+### 1.2 Start OpenSearch Stack Server
+
+Please refer to this [readme](../../vectorstores/opensearch/README.md).
+
+### 1.3 Setup Environment Variables
+
+```bash
+export OPENSEARCH_URL="http://${your_ip}:9200"
+export INDEX_NAME=${your_index_name}
+export PYTHONPATH=${path_to_comps}
+```
+
+### 1.4 Start Embedding Service
+
+First, you need to start a TEI service.
+
+```bash
+your_port=6006
+model="BAAI/bge-base-en-v1.5"
+docker run -p $your_port:80 -v ./data:/data --name tei_server -e http_proxy=$http_proxy -e https_proxy=$https_proxy --pull always ghcr.io/huggingface/text-embeddings-inference:cpu-1.5 --model-id $model
+```
+
+Then you need to test your TEI service using the following commands:
+
+```bash
+curl localhost:$your_port/embed \
+    -X POST \
+    -d '{"inputs":"What is Deep Learning?"}' \
+    -H 'Content-Type: application/json'
+```
+
+After checking that it works, set up environment variables.
+
+```bash
+export TEI_ENDPOINT="http://localhost:$your_port"
+```
+
+### 1.4 Start Document Preparation Microservice for OpenSearch with Python Script
+
+Start document preparation microservice for OpenSearch with below command.
+
+- option 1: Start single-process version (for 1-10 files processing)
+
+```bash
+cd langchain
+python prepare_doc_opensearch.py
+```
+
+<!-- - option 2: Start multi-process version (for >10 files processing)
+
+```bash
+cd langchain_ray
+python prepare_doc_opensearch_on_ray.py
+``` -->
+
+## 🚀2. Start Microservice with Docker (Option 2)
+
+### 2.1 Start OpenSearch Stack Server
+
+Please refer to this [readme](../../vectorstores/opensearch/README.md).
+
+### 2.2 Setup Environment Variables
+
+```bash
+export EMBEDDING_MODEL_ID="BAAI/bge-base-en-v1.5"
+export TEI_ENDPOINT="http://${your_ip}:6006"
+export OPENSEARCH_URL="http://${your_ip}:9200"
+export INDEX_NAME=${your_index_name}
+export HUGGINGFACEHUB_API_TOKEN=${your_hf_api_token}
+```
+
+### 2.3 Build Docker Image
+
+- Build docker image with langchain
+
+- option 1: Start single-process version (for 1-10 files processing)
+
+```bash
+cd ../../
+docker build -t opea/dataprep-opensearch:latest --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f comps/dataprep/opensearch/langchain/Dockerfile .
+```
+
+<!-- - option 2: Start multi-process version (for >10 files processing)
+
+```bash
+cd ../../../
+docker build -t opea/dataprep-on-ray-opensearch:latest --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f comps/dataprep/opensearch/langchain_ray/Dockerfile .
```

Comment:
The multi-process version here should be deleted since you haven't provide the `langchain_ray` folder.